Create contributions instead of proposals

raucao opened the pull request:
Also watch for contributions added and log them.

closes #26

Commit:
Create contributions instead of proposals

Commit:
Print actual tx error instead of guessing

bumi commented:
related to the potential noce issue: https://github.com/ethers-io/ethers.js/issues/319

Commit:
Hacky fix for nonce issues

This waits until a tx has likely been confirmed, before trying to send
the next one, because nonce management is broken in the ethers.js
wallet and it tries to replace a previous tx if two are sent quickly in
succession.

This will most likely break, when there are multiple independent
contrbution issuance calls (e.g. two PRs merged quickly in succession).
But at least it fixes issues/PRs with multiple assignees for most cases.

Commit:
"Fix" nonce issue for Mediawiki as well

See previous commit (dcbce66) for details.

raucao commented:
I think this is ready to merge, so we can start using it with the new contracts.

I opened #28 for fixing the actual nonce issue.

bumi commented:
but that "fix" does not block the http request?

raucao commented:
Nope, see the lines where I changed it to not wait for the promise to resolve.

bumi commented:
was wondering because of the await before... all that JS magic... :boom:

bumi commented on integrations/mediawiki.js:
```diff
@@ -168,10 +172,11 @@ module.exports = async function(robot, kredits) {
 
-  cron.schedule('0 7 * * *', processWikiChangesSinceLastRun);
+  // cron.schedule('0 7 * * *', processWikiChangesSinceLastRun);
+  processWikiChangesSinceLastRun();
```
I think you accidentally the `cron.schedule`

raucao commented:
There's no `await` for the handler function before returning the 200. That was wrapped in a `.then` function, which I removed. For example:

https://github.com/67P/hubot-kredits/blob/dcbce66796669c96d931a70f0f2e3ba52437af31/integrations/github.js#L173-L174

bumi reviewed (approved):
the scheduler needs to go back, then let's merge it and work on #28

Commit:
Fix accidental commenting of cron schedule

raucao commented:
Oops, I missed that comment.
